Detectors/TOF/reconstruction/src/Decoder.cxx

```diff
@@ -0,0 +1,200 @@
+#include "TOFReconstruction/Decoder.h"
+#include <iostream>
+#include <chrono>
+#include <algorithm>
+#include "CommonConstants/LHCConstants.h"
+#define VERBOSE
+
+namespace o2
+{
+namespace tof
+{
+namespace compressed
+{
+
+bool Decoder::open(std::string name)
+{
+  if (mFile.is_open()) {
+    std::cout << "Warning: a file was already open, closing" << std::endl;
+    mFile.close();
+  }
+  mFile.open(name.c_str(), std::fstream::in | std::fstream::binary);
+  if (!mFile.is_open()) {
+    std::cerr << "Cannot open " << name << std::endl;
+    return true;
+  }
+
+  if (mBuffer)
+    delete[] mBuffer;
+
+  mFile.seekg(0, mFile.end);
+  mSize = mFile.tellg();
+  mFile.seekg(0);
+
+  mBuffer = new char[mSize];
```

Comment:
mBuffer should have been deleted in the destructor. But we avoid new/delete, you can use either unique_ptr, or (easier) vector<char> datamember which you can resize here.

Reply:
replaced by vector